File: mobile/src/main/java/at/shockbytes/remote/core/SettingsActivity.java
package at.shockbytes.remote.core;

import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.transition.Explode;
import android.transition.Slide;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.Window;

import at.shockbytes.remote.fragment.SettingsFragment;

public class SettingsActivity extends AppCompatActivity{

    public static Intent newIntent(Context context) {
        return new Intent(context, SettingsActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            getWindow().requestFeature(Window.FEATURE_ACTIVITY_TRANSITIONS);
            getWindow().setExitTransition(new Slide(Gravity.BOTTOM));
            getWindow().setEnterTransition(new Explode());
        }

        if(getSupportActionBar() != null){
            getSupportActionBar().setHomeButtonEnabled(true);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        getFragmentManager()
                .beginTransaction()
                .replace(android.R.id.content, SettingsFragment.newInstance())
                .commit();
    }
	@Override
	public boolean onOptionsItemSelected(MenuItem item) {

		if (item.getItemId() == android.R.id.home) {
            supportFinishAfterTransition();
		}
		return super.onOptionsItemSelected(item);
	}

}

File: mobile/src/main/java/at/shockbytes/remote/core/SettingsActivity.kt
package at.shockbytes.remote.core

import android.content.Context
import android.content.Intent
import android.os.Bundle

import at.shockbytes.remote.dagger.AppComponent
import at.shockbytes.remote.fragment.SettingsFragment

class SettingsActivity : BackNavigableActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        fragmentManager
                .beginTransaction()
                .replace(android.R.id.content, SettingsFragment.newInstance())
                .commit()
    }

    override fun injectToGraph(appComponent: AppComponent) {
        // No injection necessary
    }

    companion object {

        val REQUEST_CODE = 0x1682

        fun newIntent(context: Context): Intent {
            return Intent(context, SettingsActivity::class.java)
        }
    }
}
